service/credential/credential.go

```diff
@@ -149,6 +113,43 @@ func GetAzureClientSetsFromCredentialSecretsBySubscription(k8sclient kubernetes.
 	return azureClientSets, nil
 }
 
+func GetAzureClientSetsByCluster(k8sclient kubernetes.Interface, g8sclient versioned.Interface) (map[string]*client.AzureClientSet, error) {
+	azureClientSets := map[string]*client.AzureClientSet{}
+	var crs []providerv1alpha1.AzureConfig
+	{
+		mark := ""
+		page := 0
+		for page == 0 || len(mark) > 0 {
+			opts := apismetav1.ListOptions{
+				Continue: mark,
+			}
+			list, err := g8sclient.ProviderV1alpha1().AzureConfigs("").List(opts)
```

Comment:
Minor note: There's actually upstream const for this.
```suggestion
			list, err := g8sclient.ProviderV1alpha1().AzureConfigs(metav1.NamespaceAll).List(opts)
```
See: https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#pkg-constants